cmd/watcher/main.go

```diff
@@ -76,6 +79,16 @@ func main() {
 	// controller's workqueue.
 	controller.DefaultThreadsPerController = *threadiness
 
+	// This parses flags.
+	k8scfg := injection.ParseAndGetRESTConfigOrDie()
+
+	if qps != nil {
+		k8scfg.QPS = float32(*qps)
```

Comment:
Should we multiply by the number of controllers we have? @vdemeester, we do something similar in pipelines.

Reply:
So see here: https://github.com/knative/pkg/blob/5d4af76051e4daa445151321f4a53953d2ce7cbb/injection/sharedmain/main.go#L241
For default, we multiply by the number of controllers. So we should multiply for values provided also.

Reply:
ah interesting ... and yeah by comparison in core tekton there is this FIXME from @vdemeester :-) - https://github.com/tektoncd/pipeline/blob/main/cmd/controller/main.go#L72-L75

so we have 2 controller at the top of the watcher and my change here switches to `MainWithConfig` per https://github.com/tektoncd/results/blob/89b591d63b025a0e7daecaff94034b32c243e6cf/cmd/watcher/main.go#L123-L129

and that there ^^  @khrm where we are now calling the `MainWithConfig` function you highlighted with your link https://github.com/knative/pkg/blob/5d4af76051e4daa445151321f4a53953d2ce7cbb/injection/sharedmain/main.go#L241 where we are already passing non-zero value for QPS and Burst means the code you highlighted, which I am now calling with this change, is doing the multiplication you are asking for.

So we are good here.  Correct?

Reply:
That code does multiplication for zero value. To have consistent behaviour across Tekton components, we should do the multiplication.
```
	if cfg.Burst == 0 {
		cfg.Burst = len(ctors) * rest.DefaultBurst
	}
```

Reply:
Ideally, knative/pkg itself should do the multiplication but I don't think that's coming anytime soon.

Reply:
yep looked too fast your right ... will update

Reply:
updated @khrm in separate commit to ease review